CONTRIBUTING.md

```diff
@@ -0,0 +1,69 @@
+This includes bug reports, fixes, documentation, examples... 
+But first, read this page.
+
+## Legal
+
+All original contributions to Narayana are licensed under the
+[GNU LESSER GENERAL PUBLIC LICENSE Version 2.1](https://github.com/jbosstm/narayana/blob/master/LICENSE).
+
+All contributions are subject to the [Developer Certificate of Origin (DCO)](https://developercertificate.org/).
+The DCO text is also included verbatim in the [dco.txt](dco.txt) file in the root directory of the repository.
+
+## Reporting an issue
+
+This project uses the [JBTM issue tracker](https://issues.redhat.com/projects/JBTM)
+to manage the issues. Open an issue directly in this issue tracker.
+
+If you believe you found a bug, and it's likely possible, please indicate a way to reproduce it, what you are seeing and what you would expect to see.
+Don't forget to indicate your Narayana, Java and Maven version. 
+
+### Building master
+
+See the [README](README.md) file in the root directory of the repository.
+
+## Before you contribute
+
+To contribute, use GitHub Pull Requests, from your **own** fork.
+
+Also, make sure you have set up your Git authorship correctly:
+
+```
+git config --global user.name "Your Full Name"
+git config --global user.email your.email@example.com
+```
+
+If you use different computers to contribute, please make sure the name is the same on all your computers.
+
+### Code reviews
+
+All submissions, including submissions by project members, need to be reviewed before being merged.
+
+### Guidelines
+
+We follow [Open source participation guidelines](https://www.redhat.com/en/resources/open-source-participation-guidelines-overview)
+
+### Coding Guidelines
+
+We use a checkstyle plugin which is skipped by default because we have a lot of old files, that we don't want to change, which would fail the compilation.
+But on new code we do encourage/advise that it should follow our checkstyle rules.
+
+To see if your code follows our style guide you have two options:
+
+Enable the checkstyle plugin via the pom file, or
+Reference the rule file from your IDE settings (for Intelij that would be Settings -> Other Settings -> Checkstyle).
+
+ * We do not use `@author` tags in the Javadoc in new code: they are hard to maintain, especially in a very active project, and we use the Git history to track authorship. GitHub also has [this nice page with your contributions](https://github.com/jbosstm/narayana/graphs/contributors).
+ * Commits should be atomic and semantic. Please properly squash your pull requests before submitting them. Fixup commits can be used temporarily during the review process but things should be squashed at the end to have meaningful commits.
+
+### Continuous Integration
+
+To ensure Narayana is stable for everyone, all changes must go through Narayana continuous integration: when you raise a pull request one of the members of the team will schedule a CI run to test your PR.
```

Comment:
again, I think we should use "should" here so this is "all changes should go through Narayana continuous integration" (even when CI is done we don't do all axis always)